syntax = "proto3";

//python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. sensor.proto



message SensorData {
    repeated float Tensao = 1;
    repeated float Corrente = 2;
    repeated float Potencia = 3;
    repeated float Energia = 4;
    repeated float FatorPot = 5;
    string Bloco = 6;
    int32 Estado = 7;
}

message DeviceInfo {
    string TIPO = 1;
    string BLOCO = 2;
    string IP = 3;
    int32 PORTA_ENVIO_TCP = 4;
}

message GatewayInfo {
    string TIPO = 1;
    int32 GTW_ID = 2;
    string IP = 3;
    int32 PORTA_ENVIO_UDP = 4;
}

message SensorDataCollection {
    repeated SensorData sensor_data = 1;
}

// Mensagem para enviar comandos ao gateway
message Command {
    enum CommandType {
        RECIEVE_DATA = 0; // Solicitar dados
        SET_STATE = 1;    // Alterar estado de um sensor
    }

    CommandType type = 1; // Tipo do comando
    string block_id = 2;  // Identificador do bloco (apenas para SET_STATE)
    bool state = 3;       // Estado desejado (apenas para SET_STATE)
}

message List {
    string list = 1;
}

package sensors;

// Definição da estrutura de um sensor
message Sensor {
  string name = 1;
  string ip = 2;
  int32 port = 3;
}

// Definição de uma lista de sensores
message SensorList {
  repeated Sensor sensors = 1;
}


message StateChange {
    string new_state = 1;
}